torchvision/transforms.py

```diff
@@ -918,6 +914,42 @@ def __call__(self, img):
         return ten_crop(img, self.size, self.vertical_flip)
 
 
+class LinearTransformation(object):
+    """Transform a tensor image with a square transformation matrix computed
+    offline.
+
+    Given transformation_matrix, will flatten the torch.*Tensor, compute the dot
+    product with the transformation matrix and reshape the tensor to its
+    original shape.
+
+    Applications:
+    - whitening: zero-center the data, compute the data covariance matrix
+                 [D x D] with np.dot(X.T, X), perform SVD on this matrix and
+                 pass it as transformation_matrix.
+
+    Args:
+        transformation_matrix (Tensor): tensor [D x D], D = C x H x W
+    """
+
+    def __init__(self, transformation_matrix):
+        assert transformation_matrix.size(0) == transformation_matrix.size(1)
```

Comment:
Could you add an error message?
Also, might be better raising a `ValueError` instead of an assert.

Reply:
I've written this

ValueError: "transformation_matrix should be square. Got [%dim0 x %dim1] rectangular matrix."

Do you have any suggestion?

Reply:
The message looks good!